mat/diagonal.go

```diff
@@ -100,6 +123,12 @@ func (d *DiagDense) TBand() Banded {
 	return TransposeBand{d}
 }
 
+// TTriBand performs an implicit transpose by returning the receiver inside a
+// TransposeTriBand. Note that Diagonal matrices are Upper by default.
+func (d *DiagDense) TTriBand() TriBanded {
+	return TransposeTriBand{d}
```

Comment:
It can just return `d`.

Reply:
The proposal would be that this is implicit, with the signature to `NewDiagonal` unchanged. It just gets modified during the call to Transpose.

Reply:
If this is needed, I'd prefer it not go in the `DiagDense`. It seems out of place there.

Reply:
I'll leave it this way for now. It's not super clear what the costs are, or that it matters that much. I think it's important to have because there are functions where you need the Triangular to be lower triangular (for instance) ad this would be otherwise impossible.

Reply:
SGTM

Reply:
To clarify, I think the approach here is better than having an `Uplo` field in the `DiagDense`.